### PR TITLE
Resolves #10 - add github action to build docker

### LINKED ISSUE #10: Add github action to build incore documentation
Review bamboo setup and migrate to github actions

### PR DESCRIPTION
**Pull request by navarroc**
Added github action workflow to build and push docker images

### CODE REVIEW

**Comment by ywkim312** [comment]
@navarroc I have a question. Maybe when we create a release, we will probably make a version number as incore doc version number. However, the tag number for api has been following the incore-services version number. In this PR, it doesn't really separates the api tag and incore tag, I guess. It shouldn't be a problem but just wanted to let you know.

**Comment by ywkim312** [comment]
Please ignore my previous comment about version number. It looks like you get the correct one and I missed that. I deleted my previous comment since it looked good.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      uses: elgohr/Publish-Docker-Github-Action@3.04
+      env:
+        BRANCH: ${{ env.GITHUB_BRANCH }}
+        VERSION: ${{ env.DOC_VERSION }}
+        BUILDNUMBER: ${{ github.run_number }}
+        GITSHA1: ${{ github.sha }}
+      with:
+        registry: hub.ncsa.illinois.edu
+        name: incore/doc/incore
+        tags: "${{ env.DOC_TAGS }}"
```
Comment (ywkim312) [inline .github/workflows/ci.yaml:89]: tags and buildargs might not needed for building the image?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,143 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - develop
```
Comment (vnarah2) [inline .github/workflows/ci.yaml:7]: we can also include 'release/*' here

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    tags:
+      - '*'
+
+  pull_request:
```
Comment (vnarah2) [inline .github/workflows/ci.yaml:12]: not mandatory, but we can also limit the PR branch list to the same in "push"

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    steps:
+    # checkout source code
+    - uses: actions/checkout@v2
+
+   # calculate some variables that are used later
+    - name: version information
+      run: |
+        if [ "${{ github.event.release.target_commitish }}" != "" ]; then
+          BRANCH="${{ github.event.release.target_commitish }}"
+        elif [[ $GITHUB_REF =~ pull ]]; then
```
Comment (vnarah2) [inline .github/workflows/ci.yaml:34]: This condition can be optimized as below so it doesn't consider the branch names with word "pull" in it
```suggestion
        elif [[ "${{github.event_name}}" == "pull_request" ]]; then
            BRANCH="PR-${{github.event.pull_request.number}}"
```

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        name: incore/doc/incore
+        username: ${{ secrets.HUB_USERNAME }}
+        password: ${{ secrets.HUB_PASSWORD }}
+        tags: "${{ env.DOC_TAGS }}"
+        buildargs: BRANCH,VERSION,BUILDNUMBER,GITSHA1
+        workdir: manual
+    # build image incore/doc/api
+    - name: Build swagger api image
+      uses: elgohr/Publish-Docker-Github-Action@3.04
+      env:
```
Comment (vnarah2) [inline .github/workflows/ci.yaml:114]: remove all envs (same for "manual")

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        BRANCH: ${{ env.GITHUB_BRANCH }}
+        VERSION: ${{ env.API_VERSION }}
+        BUILDNUMBER: ${{ github.run_number }}
+        GITSHA1: ${{ github.sha }}
+      with:
```
Comment (vnarah2) [inline .github/workflows/ci.yaml:119]: this build step only needs `workdir`, `name` and `no_push`  (same for "manual")

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        name: incore/doc/api
+        tags: "${{ env.API_TAGS }}"
+        buildargs: BRANCH,VERSION,BUILDNUMBER,GITSHA1
+        no_push: true
+        workdir: restapi
+    # This will publish to NCSA image incore/doc/api
+    - name: Publish swagger api image to NCSA Hub
+      if: github.repository == env.MAIN_REPO
+      uses: elgohr/Publish-Docker-Github-Action@3.04
+      env:
```
Comment (vnarah2) [inline .github/workflows/ci.yaml:130]: remove envs

File: .github/workflows/ci.yaml [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+        VERSION: ${{ env.API_VERSION }}
+        BUILDNUMBER: ${{ github.run_number }}
+        GITSHA1: ${{ github.sha }}
+      with:
+        registry: hub.ncsa.illinois.edu
+        name: incore/doc/api
+        username: ${{ secrets.HUB_USERNAME }}
+        password: ${{ secrets.HUB_PASSWORD }}
+        tags: "${{ env.API_TAGS }}"
+        buildargs: BRANCH,VERSION,BUILDNUMBER,GITSHA1
```
Comment (vnarah2) [inline .github/workflows/ci.yaml:141]: remove buildargs that are not used in dockerfile  (same for "manual")

**Review by vnarah2 — CHANGES_REQUESTED** [review]

**Review by vnarah2 — APPROVED** [review]

**Pull request merged**